[media-tv/plex-media-server] version bumps

ephemer0l opened the pull request:
I tweaked the HW transcode version to include some PaX marking, and included virtualenv. I've not tested it.  This also bumps the latest 'stable' with DVR support. It needed virtualenv to fix post processing abbility.

fkmclane commented:
Once again you changed the version number scheme for 1.4.3.3433 to 1.4.3 (though not for 1.4.0.3224). I don't mind it either way, but @comio changed it to that for a reason and I'll let him explain why he decided to do that. I assume it is because Plex sometimes makes releases between a single version number, like with the hardware transcoding or DVR previews.

The other stuff looks good though. I'll let @comio add his thoughts here and I'll merge the changes once the version thing is settled and the hardware transcoding preview is retested (I'll likely do this on my laptop).

ephemer0l commented:
I can understand the perspective, but I had always just bumped it with -r#.ebuild in the past. Also, the rewrite broke support on hardened which is now fixed, along with the elog message. Post processing now works with DVR, too. I expect it will be in portage soon, anyway.  I was just sharing the work done. Feel free to reversion it. I added the changes to the 1.4.0 version, too.

comio commented:
Hi ephemer0l. I added the minor version to ebuild name in order to distinguish between beta release (that keep the same version number).
Furthermore for me all solutions are fine.

ciao

luigi

comio commented:
Hi ephemer0l. I added the minor version to ebuild name in order to distinguish between beta release (that keep the same version number).
Furthermore for me all solutions are fine.

ciao

luigi

ephemer0l commented:
This should be better, then. I've made the proper adjustments for you. I also added some USE flags.

comio commented:
A 54

ephemer0l commented:
https://raw.githubusercontent.com/ephemer0l/fkmclane/master/media-tv/plex-media-server/plex-media-server-1.4.3.3433.ebuild

ephemer0l commented:
Here is a saner ebuild, as you can see I'm not the most familiar with git... I'll get it together sometime, maybe.

https://raw.githubusercontent.com/ephemer0l/fkmclane/master/media-tv/plex-media-server/plex-media-server-1.4.3.3433.ebuild

comio commented:
404: Not Found

:/

I think that we can import the commits.
I will wait for @fkmclane 's grant.

ciao

luigi

ephemer0l commented:
My sed was bad, and I should feel bad :'-(

Whats everyone's thoughts on dlna vs avahi? I used repoman to clean up some of the white space complaints, and went with 'in tree' use flags.

https://raw.githubusercontent.com/ephemer0l/overlay/master/media-tv/plex-media-server/plex-media-server-1.4.3.3433-r1.ebuild

I dont know protocol. Should I close this pull request, and open another in cases like this?

fkmclane commented on media-tv/plex-media-server/plex-media-server-1.4.3.3433-r1.ebuild:
```diff
@@ -1,16 +1,19 @@
+PYTHON_COMPAT=( python2_7 )
+inherit eutils user systemd unpacker pax-utils python-single-r1
 
+MINOR1="3433"
```
These two lines along with the commented `_FULL_VERSION` should be removed.

fkmclane commented on media-tv/plex-media-server/plex-media-server-1.4.0.3224.ebuild:
```diff
@@ -27,11 +26,11 @@ KEYWORDS="-* ~amd64"
 
-RDEPEND="plex-ffmpeg? ( >=media-video/plex-ffmpeg-2016.12 )"
+RDEPEND="plex-ffmpeg? ( >=media-video/plex-ffmpeg-2016.12 )
+	net-dns/avahi"
```
Is there any reason the Avahi USE flag isn't used here? I'm happy to add it myself; I'm just wondering if there was a reason it was omitted.

fkmclane commented:
DLNA and Avahi aren't quite comparable. DLNA specifies how UPnP should work for network media where Avahi is for service discovery. Avahi can be used to automatically discover DLNA services and PMS uses it to advertise DLNA. Since DLNA can work without Avahi, `avahi` is definitely the USE flag that should be used here.

You can just keep adding commits to your overlay and they will show up here on the pull request.

I made two comments on the review and if those are resolved, I'll merge this.

ephemer0l commented:
OK, I've merged changes between 1.4.3 into 1.4.0. Tweaked 1.4.0 to REQUIRE_USE="plex-ffmpeg' I noticed some changes in the source that they are shipping vs ffmpeg in the tree. Fixed repoman QA issues (that was the original reason I even touched the 1.4.0 build). Renamed a patch to virtualenv_start_pms.patch to match the one in portage, for compatibility/standardization. That's also why I had the commented MAJOR/MINOR to make it easier to maintain both trees.The package maintainer doesn't want plex pass ebuilds in portage, so.

I hope this all passes, and meets your standards. Thanks again for every ones time, work, and effort for Gentoo!

fkmclane commented:
It isn't exactly that they don't won't Plex Pass ebuilds so much as Plex only gave them permission for the general package.

Yeah, I just realized the version in the tree is using the MAJOR/MINOR thing so it would have been fine to leave it in. I did not notice that so that is my bad.

Thanks!

comio commented:
Regarding plex-ffmpeg, I haven't realized about the implication of REQUIRED_USE. We should remove REQUIRED_USE because we can use built-in ffmpeg (aka Plex Transcoder) insted to build from scratch.
I will commit the changeset asap.

ciao

luigi

ephemer0l commented:
They patched the source of plex-ffmpeg for at least debugging I assume.  That's why I required it.
